src/DvbData.cpp

```diff
@@ -1136,31 +1151,31 @@ void *Dvb::Process()
       {
         kodi::Log(ADDON_LOG_INFO, "Connection to the backend server failed."
           " Retrying in 10 seconds...");
-        Sleep(10000);
+        std::this_thread::sleep_for(std::chrono::milliseconds(10000));
```

Comment:
Think here makes the `kodi::tools::CThread` more sense, as by this big sleep amount it can be become problematic on his stop.

The Sleep in CThread have a special handling and a longer sleep becomes interrupted on stop.
See https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-dev-kit/include/kodi/tools/Thread.h#L305-L316

Reply:
I added a for loop to reduce the duration of the sleep.

Reply:
Erm yeah. @AlwinEsch is right. Please use a condition variable for all longer sleeps.

Reply:
I did not use a `condition_variable`. There is no way to guarantee that notify can be called while in the waiting on the variable.

I did this instead: https://github.com/kodi-pvr/pvr.dvbviewer/pull/100/commits/e27ef2bac750a009f29b88c5751cfc402f21d795